daffodil-cli/src/it/scala/org/apache/daffodil/saving/TestCLISaveParser.scala

```diff
@@ -272,173 +162,108 @@ class TestCLISaveParser {
-    shell.expect(contains("[warn] Validation Error: validation_check failed"))
-
-    cmd = Util.binPath + " save-parser --validate -s daffodil-cli/src/it/resources/org/apache/daffodil/CLI/cli_schema.dfdl.xsd -r validation_check savedParser.xml\n"
-    shell.send(cmd)
+  @Test def test_3063_CLI_Saving_SaveParser_validate(): Unit = {
+    val schema = path("daffodil-cli/src/it/resources/org/apache/daffodil/CLI/cli_schema.dfdl.xsd")
 
-    cmd2 = "echo -ne 'test'| " + Util.binPath + " parse --parser savedParser.xml \n"
-    shell.send(cmd2)
-    shell.expect(contains("[warn] Validation Error: validation_check: cvc-pattern-valid"))
-    shell.expect(contains("[warn] Validation Error: validation_check failed"))
+    withTempFile { parser =>
+      runCLI(args"save-parser --validate on -s $schema -r validation_check $parser") { cli =>
+        cli.expectErr("Unknown option 'validate'")
+      } (ExitCode.Usage)
 
-    cmd = Util.binPath + " save-parser --validate limited -s daffodil-cli/src/it/resources/org/apache/daffodil/CLI/cli_schema.dfdl.xsd -r validation_check savedParser.xml\n"
-    shell.send(cmd)
+      runCLI(args"save-parser -s $schema -r validation_check $parser") { cli =>
+      } (ExitCode.Success)
 
-    cmd2 = "echo -ne 'test'| " + Util.binPath + " parse --parser savedParser.xml \n"
-    shell.send(cmd2)
-    shell.expect(contains("[warn] Validation Error: validation_check failed"))
+      runCLI(args"parse --validate limited -P $parser") { cli =>
+        cli.send("test", inputDone = true)
+        cli.expectErr("[error] Validation Error")
+        cli.expectErr("ex:validation_check failed")
+        cli.expectErr("[0-8]+")
+      } (ExitCode.ParseError)
 
-    cmd = Util.binPath + " save-parser --validate off -s daffodil-cli/src/it/resources/org/apache/daffodil/CLI/cli_schema.dfdl.xsd -r validation_check savedParser.xml\n"
-    shell.send(cmd)
+      runCLI(args"parse --validate on -P $parser") { cli =>
+        cli.send("test", inputDone = true)
+        cli.expectErr("validation mode must be 'limited' or 'off' when using a saved parser.")
+      } (ExitCode.Usage)
+    }
+  }
 
-    cmd2 = "echo -ne 'test'| " + Util.binPath + " parse --parser savedParser.xml \n"
-    shell.send(cmd2)
+  // DAFFODIL-1141
+  /*@Test*/ def test_3036_CLI_Saving_SaveParser_debug(): Unit = {
+    val schema = path("daffodil-test/src/test/resources/org/apache/daffodil/section06/entities/charClassEntities.dfdl.xsd")
 
-    shell.send("exit\n")
-    shell.expect(eof())
-    shell.close()
-  }*/
+    withTempFile { parser =>
+      runCLI(args"-d save-parser -s $schema -r matrix $parser") { cli =>
+        cli.expectErr("Some error about -d not being valid with save-parser")
+      } (ExitCode.Usage)
+    }
+  }
 
-  // See DFDL-1141
-  /*@Test def test_3036_CLI_Saving_SaveParser_debug() {
+  // DAFFODIL-1141
+  /*@Test*/ def test_3037_CLI_Saving_SaveParser_trace(): Unit = {
```

Comment:
This PR uncomments the test's body and keeps only the `@Test` commented out.  Nice, this will keep the test compiling even after changes to functions.